worker/worker.go

```diff
@@ -209,25 +211,63 @@ func (w *Worker) handleTask(ctx context.Context, task *queue.TaskMessage) error
 		result.Status = entities.AttemptStatusFailure
 	}
 
-	result.Request = &entities.AttemptRequest{
-		URL:     request.URL,
-		Method:  request.Method,
-		Headers: utils.HeaderMap(request.Request.Header),
-		Body:    utils.Pointer(string(request.Payload)),
-	}
-	if response.StatusCode != 0 {
-		result.Response = &entities.AttemptResponse{
-			Status:  response.StatusCode,
-			Headers: utils.HeaderMap(response.Header),
-			Body:    utils.Pointer(string(response.ResponseBody)),
+	err = w.DB.TX(ctx, func(ctx context.Context) error {
```

Comment:
As an attempt only has one attempt detail. I think calling `w.DB.AttemptDetails.Upsert(attemptDetail)` after w.DB.Attempts.UpdateDelivery(ctx, task.ID, result) should be sufficient.

The SQL of Upsert would be like
--  INSERT INTO attempt_details (id, ...)  VALUES (...) ON CONFLICT (attmpt_id) DO UPDATE  SET column_1 = excluded.column_1,  column_2 = excluded.column_2;